File: solutions/rust/01-yy2/code/src/main.rs
use async_openai::{
    Client,
    config::OpenAIConfig,
    types::{ChatCompletionRequestUserMessageArgs, CreateChatCompletionRequestArgs},
};
use std::{env, error::Error, process};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let args: Vec<String> = env::args().collect();

    // Parse 'prompt' argument
    let prompt = if args.len() > 2 && args[1] == "-p" {
        args[2].clone()
    } else {
        println!("Expected usage: -p \"Your prompt\"");
        process::exit(1);
    };

    let api_key = env::var("OPENROUTER_API_KEY").map_err(|_| "OPENROUTER_API_KEY not set")?;
    let base_url = env::var("OPENROUTER_BASE_URL").map_err(|_| "OPENROUTER_BASE_URL not set")?;

    let config = OpenAIConfig::new()
        .with_api_key(api_key)
        .with_api_base(base_url);

    let client = Client::with_config(config);

    let request = CreateChatCompletionRequestArgs::default()
        .model("anthropic/claude-haiku-4.5")
        .messages([ChatCompletionRequestUserMessageArgs::default()
            .content(prompt)
            .build()?
            .into()])
        .build()?;

    let response = client.chat().create(request).await?;

    // This is to force compilation without using this variable
    // Feel free to remove this line
    let _ = response;

    if let Some(choice) = response.choices.first() {
        println!("{}", choice.message.content.as_deref().unwrap_or_default());
    }

    Ok(())
}

File: solutions/ruby/01-yy2/code/app/main.rb
require "openai"
require "optparse"

api_key = ENV["OPENROUTER_API_KEY"]
base_url = ENV.fetch("OPENROUTER_BASE_URL", "https://openrouter.ai/api/v1")

prompt = nil
OptionParser.new do |opts|
  opts.on("-p PROMPT") { |p| prompt = p }
end.parse!

raise "OPENROUTER_API_KEY is not set" unless api_key
raise "-p flag is required" unless prompt

client = OpenAI::Client.new(access_token: api_key, uri_base: base_url)

response = client.chat(
  parameters: {
    model: "anthropic/claude-haiku-4.5",
    messages: [{ role: "user", content: prompt }]
  }
)

choices = response.dig("choices")
raise "no choices in response" if choices.nil? || choices.empty?

puts choices[0].dig("message", "content")
